projects/ADuCM3029_demo_cn0428_cn0429/src/cli.c

```diff
@@ -0,0 +1,986 @@
+
+   @return none
+**/
+void CmdProcess(void)
+{
+	cmdFunc   func;
+
+	/* <ENTER> key was pressed */
+	/* Find needed function based on typed command */
+	func = FindCommand((char *)cmdInString);
+
+	/* Check if there is a valid command */
+	if (func) {
+		/* Call the desired function */
+		(*func)(&cmdInString[2]);
+
+		/* Check if there is no match for typed command */
+	} else if (strlen((const char *)cmdInString) != 0) {
+		/* Display a message for unknown command */
+		UART_TX("Unknown command!"_EOS);
+	}
+	/* Prepare for next <ENTER> */
+	// uart_cmd = UART_FALSE;
+	CmdPrompt();
+}
+
+/**
+   @brief Command line prompt. Caller must verify that only the enabled board can run this.
+
+   @return int - UART status: UART_SUCCESS or error status
+**/
+void CmdPrompt(void)
+{
+//	int res;
+	static uint8_t count = 0;
+
+
+	/* Check first <ENTER> is pressed after reset */
+	if(count == 0) {
+		UART_TX("Welcome to CN0429!"_EOS);
+		UART_TX("Type <help> to see available commands..."_EOS);
+		count++;
+	}
+
+}
+
+/**
+   @brief Find available commands
+
+   @param cmd - command to search
+
+   @return cmdFunc - return the specific function for available command or NULL
+					 for invalid command
+**/
+cmdFunc FindCommand(char *cmd)
+{
+	cmdFunc func = NULL;
+	char *p = cmd;
+	int i = 0;
+	int f1 = 0;
+	int cmdlength = 0;
+
+	while(*p != 0) {
+		if (*p == _SPC)
+			f1 = 1;
+		else if (!f1)
+			cmdlength++;
+		p++;
+	}
+
+	while (CmdFun[i] != NULL) {
+		if(strncmp(cmd, CmdCommands[i], cmdlength) == 0) {
+			func = CmdFun[i];
+			break;
+		}
+		i++;
+	}
+
+	return func;
+}
+
+void DoNothing()
+{
+	// nothing to do here
+	// this is to avoid help printing each time enter is pressed
+}
+
+/**
+   @brief Set sensor to use for single sensor commands
+
+   @param args - pointer to the arguments on the command line.
+   	   site: 1, 2, 3, or 4
+
+   @return none
+**/
+void SetDefaultSns(uint8_t *args)
+{
+	uint8_t  *p = args;
+	char     arg[17];
+	char	 buff[20] = {0};
+
+	/* Check if this function gets an argument */
+	while (*(p = FindArgv(p)) != '\0') {
+		/* Save channel parameter */
+		GetArgv(arg, p);
+	}
+
+	/*DEFAULT CONFIGURATION*/
+	if (strncmp(arg, "1", 2) == 0) {
+		uiDefaultSite = 1;
+		uiDefaultAddress = ADI_CFG_I2C_SENSOR1_ADDR;
+	} else if (strncmp(arg, "2", 2) == 0) {
+		uiDefaultSite = 2;
+		uiDefaultAddress = ADI_CFG_I2C_SENSOR2_ADDR;
+	} else if (strncmp(arg, "3", 2) == 0) {
+		uiDefaultSite = 3;
+		uiDefaultAddress = ADI_CFG_I2C_SENSOR3_ADDR;
+	} else if (strncmp(arg, "4", 2) == 0) {
+		uiDefaultSite = 4;
+		uiDefaultAddress = ADI_CFG_I2C_SENSOR4_ADDR;
+	} else {
+		UART_TX("Invalid site"_EOS);
+	}
+	sprintf(buff, "Selected site: %d%s", uiDefaultSite, _EOS);
+	UART_TX((const char*)buff);
+}
+
+/**
+   @brief Print which sensors are connected
+
+   @param args - pointer to the arguments on the command line.
+
+   @return none
+**/
+void SnsConnected(uint8_t *args)
+{
+	memset(strCmd,'\0',sizeof(strCmd));
```

Comment:
These functions could actually do the processing and not just add strings to the ring buffer. This will eliminate the need for the ring buffer and reduce redundancy.

Reply:
Addressed this above also: https://github.com/analogdevicesinc/EVAL-ADICUP3029/pull/12#discussion_r243325209